drivers/console/bm_serial.c

```diff
@@ -356,28 +124,60 @@ static void bm_serial_rx_thread(void)
 	uint16_t received_crc16;
 	int retval;
 	int i;
+
 	/* Buffer to store Manchester Decoded RX data */
 	uint8_t man_decode_buf[ MAX_BM_FRAME_SIZE ] = {0};
-	uint8_t man_decode_len = 0;
+	uint16_t man_decode_len = 0;
 	int8_t decoded_nibble = 0;
+	uint8_t preamble_err = 0;
 
 	LOG_DBG("BM Serial RX thread started");
 
+	bm_dec_msg_t msg;
+	const struct device *dev;
+
 	while (1)
 	{
 		/* Wait on Producer to finish writing out decoded frame */
 		k_sem_take(&decode_sem, K_FOREVER);
 
+		/* Check if length is even */
+		if (encoded_rx_buf[read_buf_idx].length & 0x01)
```

Comment:
Is `encoded_rx_buf` the same buffer DMA is writing to?
If so, we might consider making a copy of that buffer so DMA can keep doing it's thing ASAP.

I know we're ping-ponging, but if this decoding thread takes too long for whatever reason (other things going on in the system), DMA will happily overwrite the contents of the buffer and you'll end up decoding two different messages (thinking it's one)

Reply:
No, the buffers that DMA write to are dev_ctx[i].buf (1 for each serial interface). In the cb function, the ISR memcpys the value from these buffers into the encoded_rx_buf ping pong buffer.